turbopuffer/query.py

```diff
@@ -37,50 +40,67 @@
 
 @dataclass
 class VectorQuery:
-    vector: Optional[List[float]] = None
-    distance_metric: Optional[str] = None
+    rank_by: RankInput
     top_k: int = 10
-    include_vectors: bool = False
     include_attributes: Optional[Union[List[str], bool]] = None
     filters: Optional[Filters] = None
-    rank_by: Optional[RankInput] = None
     consistency: Optional[ConsistencyDict] = None
-    vector_encoding_format: Optional[str] = None
+    vector_encoding: Optional[str] = None
 
     def from_dict(source: dict) -> 'VectorQuery':
         return VectorQuery(
-            vector=source.get('vector'),
-            distance_metric=source.get('distance_metric'),
             top_k=source.get('top_k'),
-            include_vectors=source.get('include_vectors'),
             include_attributes=source.get('include_attributes'),
             filters=source.get('filters'),
             rank_by=source.get('rank_by'),
             consistency=source.get('consistency'),
-            vector_encoding_format=source.get('vector_encoding_format'),
+            vector_encoding=source.get('vector_encoding'),
         )
 
     def __post_init__(self):
-        if self.vector is not None:
-            if 'numpy' in sys.modules and isinstance(self.vector, sys.modules['numpy'].ndarray):
-                if self.vector.ndim != 1:
-                    raise ValueError(f'VectorQuery.vector must be a 1d array, got {self.vector.ndim} dimensions')
-            elif not isinstance(self.vector, list):
-                raise ValueError('VectorQuery.vector must be a list, got:', type(self.vector))
+        if self.rank_by is None:
+            raise ValueError('VectorQuery.rank_by is required')
+        if len(self.rank_by) == 3 and self.rank_by[:2] == ['vector', 'ANN']:
+            vector = self.rank_by[2]
+            if 'numpy' in sys.modules and isinstance(vector, sys.modules['numpy'].ndarray):
+                if vector.ndim != 1:
+                    raise ValueError(f'VectorQuery.vector must be a 1d array, got {vector.ndim} dimensions')
```

Comment:
i think this is meant to be the following?
```suggestion
                    raise ValueError(f'VectorQuery.rank_by vector must be a 1d array, got {vector.ndim} dimensions')
```

Reply:
Oh wow, good eye! Yes, thank you!